packages/typespec-autorest/src/openapi.ts

```diff
@@ -2373,9 +2378,18 @@ export async function getOpenAPIForService(
           visibility: context.visibility | Visibility.Item,
         }),
       };
-      if (!ifArrayItemContainsIdentifier(program, typespecType as any)) {
+
+      const armIdentifiers = getArmIdentifiers(program, typespecType);
```

Comment:
Note that name and id are considered valid as identifiers, even if they are not marked with `@key`